### PR TITLE
Add local vertex map for arrow fragment

### PR DESCRIPTION
**Pull request by acezen**
What do these changes do?
-------------------------

Implement local vertex map for arrow fragment 

Related issue number
--------------------

Fixes https://github.com/alibaba/GraphScope/issues/1689

### CODE REVIEW

**Comment by acezen** [comment]
cc/ @sighingnow @siyuan0322

File: modules/graph/vertex_map/arrow_vertex_map.h [hunk 1]
```diff
@@ -303,8 +296,6 @@ class ArrowVertexMap
   std::vector<std::vector<vineyard::Hashmap<oid_t, vid_t>>> o2g_;
 
   friend class ArrowVertexMapBuilder<OID_T, VID_T>;
-
-  friend class gs::ArrowProjectedVertexMap<OID_T, VID_T>;
 };
```
Comment (sighingnow) [inline modules/graph/vertex_map/arrow_vertex_map.h:299]: Remove the friend class declaration may break graphscope. Please double check the usages.

File: modules/graph/loader/basic_ev_fragment_loader.h [hunk 2]
```diff
@@ -88,7 +93,7 @@ class BasicEVFragmentLoader {
   /// Set attributes: vertex_label_num_, vm_ptr_, output_vertex_tables_
   ///                 vertex_label_to_index_, vertex_labels_
   boost::leaf::result<void> ConstructVertices(
-      ObjectID vm_id = InvalidObjectID()) {
+      ObjectID vm_id = InvalidObjectID(), bool with_local_vertex_map = false) {
```
Comment (sighingnow) [inline modules/graph/loader/basic_ev_fragment_loader.h:96]: We already have a type paramter `VERTEX_MAP_T`, why the runtime paramter `with_local_vertex_map` is still needed?

**Review by sighingnow — APPROVED** [review]

**Review by siyuan0322 — APPROVED** [review]

File: modules/graph/loader/basic_ev_fragment_loader.h (hunk 2 above)
Comment (acezen) [reply modules/graph/loader/basic_ev_fragment_loader.h:96]: the `LocalVertexMap` construct logic is not same with `GlobalVertexMap`, and it's not easy to unify them.

File: modules/graph/vertex_map/arrow_vertex_map.h (hunk 1 above)
Comment (acezen) [reply modules/graph/vertex_map/arrow_vertex_map.h:299]: I refine the `ProjectedVetexMap` in https://github.com/alibaba/GraphScope/pull/1794 and it's no need to access vertex_map's private member, so we can remove the friend declaration here.

**Pull request closed without merging**